CMakeLists.txt

```diff
@@ -1,17 +1,28 @@
 cmake_minimum_required(VERSION 3.22)
-project(NEBULA)
+project(Nebula)
 
-set(CMAKE_CXX_STANDARD 11)
-
-set(CMAKE_CXX_COMPILER "g++")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -lGL")
+set(CMAKE_CXX_STANDARD 20)
+# Change to ON/OFF depending on whether you want to use clang instead of g++
+option(USE_CLANG "Use clang instead of g++" OFF)
+if(USE_CLANG)
+    message(STATUS "Using clang as compiler")
+    set(CMAKE_CXX_COMPILER "clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++20 -Weverything -Wno-c++98-compat")
+else()
+    # Use g++ by default
+    message(STATUS "Using g++ as compiler")
+    set(CMAKE_CXX_COMPILER "g++")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++20 -Wall -Wextra -pedantic")
```

Comment:
-std=c++20 will be set by cmake. setting it manually prevents the project from being configured by cmake